pkg/resourcemanager/vm/reconcile.go

```diff
@@ -91,10 +101,19 @@ func (g *AzureVirtualMachineClient) Ensure(ctx context.Context, obj runtime.Obje
 			switch azerr.Type {
 			case errhelp.AsyncOpIncompleteError:
 				instance.Status.Provisioning = true
+				instance.Status.Message = err.Error()
+
 			}
 			// reconciliation is not done but error is acceptable
 			return false, nil
 		}
+		//change to spec is required
+		if future.Response().StatusCode == 400 {
```

Comment:
Some SDK clients don't return full future objects on failures. Have you tested any failure scenarios here to ensure that Response() doesn't cause a nil pointer dereference here?

Reply:
Seems like compute sdk returns with status code because i can get the 400's. This addresses [#1090]